sdk/src/main/java/com/uid2/sdk/UID2Client.kt

```diff
@@ -0,0 +1,72 @@
+package com.uid2.sdk
+
+import com.uid2.sdk.network.DataEnvelope
+import com.uid2.sdk.network.NetworkRequest
+import com.uid2.sdk.network.NetworkRequestType
+import com.uid2.sdk.network.NetworkSession
+import com.uid2.sdk.network.RefreshPackage
+import com.uid2.sdk.network.RefreshResponse
+import java.net.HttpURLConnection
+import java.net.URI
+import java.net.URL
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.json.JSONObject
+
+/**
+ * This class is responsible for refreshing the identity, using a provided refresh token. The payload response will be
+ * encrypted, so also is provided a key to allow decryption.
+ */
+class UID2Client(
+    private val apiUrl: String,
+    private val session: NetworkSession
+) {
+    // The refresh endpoint is built from the given API root, along with our known refresh path appended. If the
+    // consumer has incorrectly configured the SDK, it's possible this could be null.
+    private val apiRefreshUrl: URL? by lazy {
+        runCatching {
+            URL(URI(apiUrl)
+                .resolve(API_REFRESH_PATH)
+                .toString()
+            )
+        }.getOrNull()
+    }
+
+    suspend fun refreshIdentity(
```

Comment:
This won't be part of the public API, and just used internally by the `UID2Manager` (to be implemented). My thought is that the manager internally will have it's own `CoroutineScope` around the user authentication. This scope will be used by the refresh timer, and when requiring to call this `suspend` function.